lib/instances/BaseInstance.lua

```diff
@@ -157,9 +157,18 @@ function BaseInstance.prototype:GetChildren()
 end
 
 function BaseInstance.prototype:IsA(className)
-	-- TODO: Hierarchy stuff
+	local currentClass = getmetatable(self).class
 
-	return self.ClassName == className
+	while (currentClass ~= nil)
+	do
```

Comment:
For style, let's put `do` on the same line as the `while` bit and remove these parentheses.